### PR TITLE
Rails 4

### PR DESCRIPTION
**Pull request by Fosome**
Make this branch backwards compatible with Rails 3 stuffs.

### CODE REVIEW

File: lib/simplest_auth/active_model/model.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+unless defined?(ActiveModel::Model)
+
+  # Ripped from ActiveModel::Model 4.1.1
+  # https://github.com/rails/rails/blob/v4.1.1/activemodel/lib/active_model/model.rb
```
Comment (efatsi) [inline lib/simplest_auth/active_model/model.rb:4]: want to throw a commit hash in this url so it's always findable in the future? i hate it when these kinds of links 404 because i typically really need it if I'm clicking on it.

Comment (brianjlandau) [inline lib/simplest_auth/active_model/model.rb:4]: This is using a tag, why would the tag go away? I'm not seeing how this would 404 in the future.

File: lib/simplest_auth/active_model/model.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+unless defined?(ActiveModel::Model)
```
Comment (efatsi) [inline lib/simplest_auth/active_model/model.rb:1]: why is this needed? seems odd that this could be undefined but the things on lines 10-14 would be.

**Comment by brianjlandau** [comment]
Looks good :+1: but I'd consider adding [Appraisal](https://github.com/thoughtbot/appraisal) to test against a number of rails versions.

File: lib/simplest_auth/bcrypt.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+begin
+  require 'bcrypt'
+rescue LoadError
+  begin
+    gem 'bcrypt-ruby'
+  rescue Gem::LoadError
+    puts "Please install the bcrypt-ruby gem"
+  end
+end
```
Comment (efatsi) [inline lib/simplest_auth/bcrypt.rb:9]: can you just remove this since bcrypt-ruby is a dependency?

File: spec/lib/model/data_mapper_spec.rb [hunk 4]
```diff
@@ -1,5 +1,10 @@
 require 'spec_helper'
-require 'datamapper'
+
+begin
+  require 'data_mapper'
+rescue LoadError
+  require 'datamapper'
+end
```
Comment (efatsi) [inline spec/lib/model/data_mapper_spec.rb:7]: wat?

File: lib/simplest_auth/active_model/model.rb (hunk 2 above)
Comment (brianjlandau) [inline lib/simplest_auth/active_model/model.rb:1]: The items on lines 10-14 are defined as far back as version 3 https://github.com/rails/rails/blob/v3.0.0/activemodel/lib/active_model/translation.rb however `ActiveModel::Model` is new to rails 4.

File: lib/simplest_auth/active_model/model.rb (hunk 1 above)
Comment (efatsi) [inline lib/simplest_auth/active_model/model.rb:4]: ah didn't know the version param there did the same thing, sorry ignore me

**Comment by Fosome** [comment]
@brianjlandau Yup.  I'm going to put a testing matrix in place before I merge.

**Comment by efatsi** [comment]
left some useless comments for you. :+1:

File: lib/simplest_auth/bcrypt.rb (hunk 3 above)
Comment (Fosome) [inline lib/simplest_auth/bcrypt.rb:9]: I'm not sure?  Just moving this around, but worth looking into.

File: spec/lib/model/data_mapper_spec.rb (hunk 4 above)
Comment (Fosome) [inline spec/lib/model/data_mapper_spec.rb:7]: right?  see https://twitter.com/ry_fo/status/469164558427648000

**Comment by Fosome** [comment]
Thanks guise
